File: baekjoon/gold/4_11404_플로이드/플로이드.js
const fs = require("fs");
const file = process.platform === "linux" ? "/dev/stdin" : "./text.txt";
const texts = fs.readFileSync(file).toString().trim().split("\n");

const n = texts.shift();
const m = texts.shift();
const INF = Number(Infinity);
let graph = Array.from({ length: n + 1 }, () => Array(n + 1).fill(INF));

for (let a = 1; a < n + 1; a++) {
  for (let b = 1; b < n + 1; b++) {
    if (a === b) {
      graph[a][b] = 0;
    }
  }
}
for (const [a, b, c] of texts) {
  if (graph[a][b] > c) {
    graph[a][b] = c;
  }
}

for (let k = 1; k < n + 1; k++) {
  for (let a = 1; a < n + 1; a++) {
    for (let b = 1; b < n + 1; b++) {
      graph[a][b] = Math.min(graph[a][b], graph[a][k] + graph[k][b]);
    }
  }
}
for (let i = 1; i < n + 1; i++) {
  console.log(
    graph[i]
      .slice(1)
      .map((value) => {
        if (value === INF) {
          return [0];
        }
        return value;
      })
      .flat()
      .join(" ")
  );
}

File: baekjoon/gold/4_11404_플로이드/플로이드.py
# https://www.acmicpc.net/problem/11404
# n개 도시가 존재 A -> B로 가는 비용의 최솟값
import sys
input = sys.stdin.readline

n = int(input())
m = int(input())
INF = int(1e9)
graph = [[INF] * (n+1) for _ in range(n+1)]


for a in range(1, n+1):
  for b in range(1, n+1):
    if a == b:
      graph[a][b] = 0

      
for _ in range(m):
  a, b, c = map(int, input().split())
  if graph[a][b] > c:
    graph[a][b] = c


for k in range(1, n+1):
  for a in range(1, n+1):
    for b in range(1, n+1):
      graph[a][b] = min(graph[a][b], graph[a][k] + graph[k][b])

for i in range(1, n+1):
  for j in range(1, n+1):
    if graph[i][j] >= INF:
      print(0, end=" ")
    else:
      print(graph[a][b], end=" ")
  print()